### PR TITLE
[CatalogPromotions] Showing catalog promotions in Products index and show page

### PR DESCRIPTION
**Pull request by SirDomin**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

Products index:

Zrzut ekranu 2021-09-15 o 14 17 50

Product's details:

Zrzut ekranu 2021-09-15 o 14 17 38

### CODE REVIEW

File: src/Sylius/Behat/Context/Ui/Shop/ProductContext.php [hunk 1]
```diff
@@ -351,8 +351,19 @@ public function iShouldSeeTheProductOriginalPrice($price)
...
     }
 
+    /**
+     * @Then /^I should see ("[^"]+" variant) is not discounted$/
+     */
+    public function iShouldSeeVariantIsNotDiscounted(ProductVariantInterface $variant): void
+    {
+        $this->showPage->selectVariant($variant->getName());
+
+        Assert::same($this->showPage->getPrice(), $this->showPage->getOriginalPrice());
```
Comment (AdamKasp) [inline src/Sylius/Behat/Context/Ui/Shop/ProductContext.php:361]: are you sure about this? from the shop UI perspective if `originalPrice` is not set, the customer sees only `price` not two same value.

Comment (SirDomin) [reply src/Sylius/Behat/Context/Ui/Shop/ProductContext.php:361]: but its label is still visible, i wasnt able to check if originalPrice is not visible as it always returns true :/ thats the only way i was able to get if the promotion is applied

File: src/Sylius/Bundle/ShopBundle/Resources/views/Product/Show/_price.html.twig [hunk 2]
```diff
@@ -2,6 +2,20 @@
 
 {% set variant = product|sylius_resolve_variant %}
 {% set hasDiscount = variant|sylius_has_discount({'channel': sylius.channel}) %}
+{% set channelPricing = variant.channelPricings.first %}
```
Comment (AdamKasp) [inline src/Sylius/Bundle/ShopBundle/Resources/views/Product/Show/_price.html.twig:5]: you shoudn't get here first `channelPricing` but channel pricing for active channel 
```suggestion
{% set channelPricing = variant.getChannelPricingForChannel(sylius.channel %}
```

**Review by AdamKasp — APPROVED** [review]

File: src/Sylius/Behat/Context/Api/Shop/ProductContext.php [hunk 3]
```diff
@@ -165,6 +165,16 @@ public function iShouldSeeTheProductOriginalPrice(int $originalPrice): void
...
     }
 
+    /**
+     * @Then I should see this product has no catalog promotion applied
+     */
+    public function iShouldSeeThisProductHasNoCatalogPromotionApplied(): void
+    {
+        $variant = $this->responseChecker->getResponseContent($this->client->getLastResponse());
+
+        Assert::same($variant['originalPrice'], $variant['price']);
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Shop/ProductContext.php:175]: You can also check if `appliedPromotions` field is empty

File: src/Sylius/Behat/Context/Ui/Shop/ProductContext.php (hunk 1 above)
Comment (GSadee) [reply src/Sylius/Behat/Context/Ui/Shop/ProductContext.php:361]: I don't understand why the label is still visible if there is no applied promotion and why it always return true?

File: src/Sylius/Behat/Page/Shop/Product/ShowPage.php [hunk 4]
```diff
@@ -309,6 +314,7 @@ protected function getDefinedElements(): array
             'attributes' => '[data-test-product-attributes]',
             'average_rating' => '[data-test-average-rating]',
             'current_variant_input' => '[data-test-product-variants] td input:checked',
+            'catalog_promotion_name' => '#sylius_catalog_promotion_name',
```
Comment (GSadee) [inline src/Sylius/Behat/Page/Shop/Product/ShowPage.php:317]: Wrong order of keys

File: src/Sylius/Behat/Resources/config/suites.yml [hunk 5]
```diff
@@ -104,6 +104,7 @@ imports:
     - suites/ui/product/viewing_product_reviews.yml
     - suites/ui/product/viewing_products.yml
     - suites/ui/promotion/applying_promotion_coupon.yml
+    - suites/ui/promotion/applying_catalog_promotions.yml
```
Comment (GSadee) [inline src/Sylius/Behat/Resources/config/suites.yml:107]: Wrong order of suites

File: features/promotion/applying_catalog_promotions/seeing_applied_catalog_promotions_on_variants_list.feature [hunk 6]
```diff
@@ -12,9 +12,10 @@ Feature: Seeing applied catalog promotions on variants list
         And this product has "Python T-Shirt" variant priced at "$40.00"
         And there is a catalog promotion "Winter sale" that reduces price by "50%" and applies on "PHP T-shirt" variant and "Python T-Shirt" variant
 
-    @api
+    @api @ui @javascript
     Scenario: Seeing applied catalog promotion on variant
-        When I view variants
+        When I view product "T-Shirt"
```
[the remark this answers is not in the document]
Comment (Zales0123) [reply features/promotion/applying_catalog_promotions/seeing_applied_catalog_promotions_on_variants_list.feature:17]: Indeed 👍 in fact, I've created a separated feature file fo that, as there is no such thing as "variants list" in the UI context

File: src/Sylius/Behat/Context/Ui/Shop/ProductContext.php (hunk 1 above)
Comment (Zales0123) [reply src/Sylius/Behat/Context/Ui/Shop/ProductContext.php]: fixed 💪

File: features/promotion/applying_catalog_promotions/seeing_applied_catalog_promotion_on_variant.feature [hunk 7]
```diff
@@ -8,9 +8,17 @@ Feature: Seeing applied catalog promotions on variant
...
-        When I view "PHP T-Shirt" variant
-        Then I should see "PHP T-Shirt" variant is discounted from "$20.00" to "$10.00" with "Winter sale" promotion
+        When I view product "T-Shirt"
+        And I view "PHP T-Shirt" variant
+        Then I should see this variant is discounted from "$20.00" to "$10.00" with "Winter sale" promotion
+
+    @api @ui @javascript
+    Scenario: Seeing no applied catalog promotion on variant
+        When I view product "T-Shirt"
+        And I view "Sylius T-Shirt" variant
```
Comment (GSadee) [inline features/promotion/applying_catalog_promotions/seeing_applied_catalog_promotion_on_variant.feature:23]: Maybe sth like:
```suggestion
        When I view "Sylius T-Shirt" variant of a "T-Shirt" product
```

File: features/promotion/applying_catalog_promotions/seeing_applied_catalog_promotions_on_variants_list.feature [hunk 8]
```diff
@@ -12,9 +12,10 @@ Feature: Seeing applied catalog promotions on variants list
         And this product has "Python T-Shirt" variant priced at "$40.00"
         And there is a catalog promotion "Winter sale" that reduces price by "50%" and applies on "PHP T-shirt" variant and "Python T-Shirt" variant
 
-    @api
+    @api @no-ui
```
Comment (GSadee) [inline features/promotion/applying_catalog_promotions/seeing_applied_catalog_promotions_on_variants_list.feature:15]: New tag 😱, but 👌🏻

File: features/promotion/applying_catalog_promotions/seeing_applied_catalog_promotions_on_variants_list.feature [hunk 9]
```diff
@@ -12,9 +12,10 @@ Feature: Seeing applied catalog promotions on variants list
...
     Scenario: Seeing applied catalog promotion on variant
-        When I view variants
+        When I view product "T-Shirt"
+        And I view variants
```
Comment (GSadee) [inline features/promotion/applying_catalog_promotions/seeing_applied_catalog_promotions_on_variants_list.feature:18]: Can be merged to one step

File: features/promotion/applying_catalog_promotions/seeing_applied_catalog_promotions_on_variants_list.feature (hunk 8 above)
Comment (Zales0123) [reply features/promotion/applying_catalog_promotions/seeing_applied_catalog_promotions_on_variants_list.feature:15]: XD

**Review by AdamKasp — APPROVED** [review]

**Comment by AdamKasp** [comment]
Thank you, @SirDomin! :tada:

**Pull request merged**